index.js

```diff
@@ -119,7 +103,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^([^]+)\/([^]+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (Nullable String, String, Number)
+  //                      -> { namespace :: Nullable String
+  //                         , name :: String
+  //                         , version :: Number }
+  function TypeIdentifier(namespace, name, version) {
+    return {namespace: namespace, name: name, version: version};
+  }
+
+  //. ### Usage
+  //.
+  //. `const {parse, type} = require('sanctuary-type-identifiers')`
+  //.
+  //. ```javascript
+  //. > parse(type([1, 2, 3]));
+  //. {namespace: null, name: 'Array', version: 0}
+  //. ```
+  //.
+  //. ### API
+  //.
+  //# type :: Any -> String
+  //.
+  //. Takes any value and produces a String which represents its type. If the
+  //. value conforms to the [specification][3], the custom type identifier is
+  //. returned.
+  //.
+  //. ```javascript
+  //. > type(null);
+  //. 'Null'
+  //.
+  //. > type(true);
+  //. 'Boolean'
```

Comment:
I like to omit trailing semicolons when writing doctests. Doctests are meant to resemble REPL interactions, after all. :)